src/index/base.cpp

```diff
@@ -160,12 +160,24 @@ void BaseIndex::Sync()
             }
 
             const CBlockIndex* pindex_next = WITH_LOCK(cs_main, return NextSyncBlock(pindex, m_chainstate->m_chain));
+            // If pindex_next is null, it means pindex is the chain tip, so
+            // commit data indexed so far.
```

Comment:
nit: I had to read this three times to get it right, but feel free to ignore unless you have to retouch.

I would change it to `[...], so commit the data that was indexed so far.`